packages/devextreme/testing/testcafe/runner.js

```diff
@@ -16,6 +16,22 @@ const changeTheme = async(themeName) => createTestCafe.ClientFunction(() => new
 }),
 { dependencies: { themeName } })();
 
+const addShadowRootTree = async function() {
+    await createTestCafe.ClientFunction(() => {
+        const root = document.querySelector('#parentContainer');
+        const childNodes = root.childNodes;
+
+        if(!root.shadowRoot) {
+            root.attachShadow({ mode: 'open' });
+        }
+
+        const shadowContainer = document.createElement('div');
+        shadowContainer.append.apply(shadowContainer, Array.from(childNodes));
```

Comment:
How about just shadowContainer.append(Array.from(childNodes)); ?

Reply:
The [append](https://developer.mozilla.org/en-US/docs/Web/API/Element/append) method does not support array as argument